toolchain/check/decl_name_stack.h

```diff
@@ -151,6 +151,10 @@ class DeclNameStack {
       // The ID of an unresolved identifier.
       SemIR::NameId unresolved_name_id = SemIR::NameId::None;
     };
+
+    // When `state` is `Poisoned` (name is unresolved due to name poisoning),
+    // the poisoning location.
+    SemIR::LocId poisoning_loc_id;
```

Comment:
```suggestion
    SemIR::LocId poisoning_loc_id = SemIR::LocId::None;
```

Noted in the cpp file too, but I think using a default instead of always initializing to `None` would be consistent with other member init.

Reply:
Done.